pkg/ruler/api.go

```diff
@@ -154,6 +155,14 @@ func (a *API) PrometheusRules(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	typ := strings.ToLower(req.URL.Query().Get("type"))
+	if typ != "" && typ != "alert" && typ != "record" {
+		respondError(logger, w, fmt.Sprintf("not supported value %q", typ))
```

Comment:
`respondError` will return a HTTP 500 response, but we should probably return a 400 (bad request) here.

Reply:
Yeah, I wanted to do that initially, but then I saw that missing user ID is returned same way as 500 error being more of an invalid request also, so I just followed the lead instead of adding more code for marshalling the JSON, but sure I'll add the correct response as Prometheus does :+1:

Reply:
I had to abstract the `returnError` function a bit to avoid code duplication. Hope it's ok like this.
See https://github.com/grafana/mimir/pull/4302/commits/6f40ca3f32d0298754466351e47e448bb2ec1da1